classes/submission/genre/Genre.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace PKP\submission\genre;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\submissionFile\SubmissionFile;
+
+
+
+class Genre extends Model
+{
+    const GENRE_CATEGORY_DOCUMENT = 1;
+    const GENRE_CATEGORY_ARTWORK = 2;
+    const GENRE_CATEGORY_SUPPLEMENTARY = 3;
+    protected $table = 'genres';
+    protected $primaryKey = 'genre_id';
+    public $timestamps = false;
+
+    protected $fillable = [
+        'context_id', 'seq', 'enabled', 'category', 'dependent', 
+        'supplementary', 'required', 'entry_key'
+    ];
+
+    // Relationships
+    public function submissionFiles(): HasMany
+    {
+        return $this->hasMany(SubmissionFile::class, 'genre_id');
+    }
+
+
+    // Accessors and Mutators
+    protected function entryKey(): Attribute
+    {
+        return Attribute::make(
```

Comment:
Was there an equivalent to this in the previous implementation?

Reply:
It's the uppercase/lowercase conversion I was wondering about; if it didn't exist in the previous code, best not to add it here.